CONTRIBUTING.md

```diff
@@ -109,42 +115,346 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+    * Sections
+* **Comment styles**
+    * Recommend to add the “/** */” (Javadoc) style comments to all interface methods, with explanation of the parameters and returns. E.g.
+
+      ```java
+          interface Transformation<T, R>
+          {
+              /**
+               * Processes input elements and returns current transformation state.
+               *
+               * @param elementOptional an element to be transformed. Will be empty
+               * when there are no more elements. In such case transformation should
+               * finish processing and flush any remaining data.
+               * @return the current transformation state, optionally bearing a result
+               * @see TransformationState#needsMoreData()
+               * @see TransformationState#blocked(ListenableFuture)
+               * @see TransformationState#yield()
+               * @see TransformationState#ofResult(Object)
+               * @see TransformationState#ofResult(Object, boolean)
+               * @see TransformationState#finished()
+               */
+              TransformationState<R> process(Optional<T> elementOptional);
+          }
+      ```
+
+    * Recommend to add the “/** */” style comments to important or difficult public methods, with explanation of the parameters and returns.
+    * Within the method body, use “//” style comments on separate lines for difficult parts.
+    * Use “//” style comments for class members if it can help understanding the code. Can be on the same line or separate lines in front.
+* **Code succinctness**
+    * Inline function calls when appropriate. E.g. if a function is only called once, we don’t need to create a variable for it.
+    * Lambda
+      E.g. Instead of doing this
+
+      ```java
+      Iterables.getOnlyElement(argumentSets).stream()
+      .map((rowExpression) -> OriginalExpressionUtils.castToExpression(rowExpression))
+      ```
+      We just do
+      ```java
+      Iterables.getOnlyElement(argumentSets).stream()
+      .map(OriginalExpressionUtils::castToExpression)
+      ```
+
+* When appropriate use Java 8 Stream API
+* Categorize errors when throwing an exception
+
+
+## Commit Standards
+* ### Commit Size
+    * Recommended lines of code should be within +1000 lines, and should focus on one single major topic.\
+      If your commit is more than 1000 lines, consider breaking it down into multiple commits, each handling a specific small topic.
+* ### Commit Message Style
+    * **Separate subject from body with a blank line**
+    * **Subject**
+        * Limit the subject line to 10 words or 50 characters
+        * If you cannot make the subject short, you may be committing too many changes at once
+        * Capitalize the subject line
+        * Do not end the subject line with a period
+        * Use the imperative mood in the subject line
+    * **Body**
+        * Wrap the body at 72 characters
+        * Use the body to explain what and why vs. how
```

Comment:
```suggestion
        * Use the body to explain what and why versus how
```
GitLab doc style guide doesn't mention vs. or versus. The [Google developer doc word list entry for vs.](https://developers.google.com/style/word-list#vs) recommends using the unabbreviated version.